app/src/controllers/enquiry.ts

```diff
@@ -178,6 +178,22 @@ const controller = {
     } catch (e: unknown) {
       next(e);
     }
+  },
+
+  validateActivityId: async (req: Request<{ activityId: string }>, res: Response, next: NextFunction) => {
+    try {
+      const { activityId } = req.params;
+      const hexidecimal = parseInt(activityId, 16);
+
+      if (activityId.length !== 8 || !hexidecimal) {
+        return res.status(400).json({ message: 'Invalid activity Id format' });
+      }
+      const activity = await activityService.getActivity(activityId);
+
+      res.status(200).json({ valid: Boolean(activity && !activity.isDeleted) });
```

Comment:
Is the Boolean wrapper necessary here? `activity && !activity.isDeleted` should evaluate to what you're looking for.

Reply:
If boolean wrapping is required if/when one of the variables ends up being undefined/falsy, it would be better do do it with a double `!!` as we typically want to avoid using JS boxing functions when we are forced to cast to a different type.
```ts
res.status(200).json({ valid: !!activity && !activity.isDeleted });
```